### PR TITLE
feat: auto publish to npm

### PR DESCRIPTION
**Pull request by ganchoradkov**
## Description
- Added bash script to set `web3wallet` sdk version automatically relative to the main monorepo version
- Added workflow to auto publish to npm when new release is created

TODO: add npm token secret to the repo

## New release flow
1. npm run pre-release
2. git tag `<the version>`
3. push the tag
4. create release on gh

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
bash script tested, auto publish todo

## Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Additional Information (Optional)

Please include any additional information that may be useful for the reviewer.

### CODE REVIEW

**Comment by jakubuid** [comment]
@ganchoradkov 
Does this update Core, Relayer, React Native Compat, Sign Client, Types, Ethereum Provider, Universal Provider, Signer Connection as well?

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "test:ignoreUnhandled": "npm run test:ignoreUnhandled --workspaces --if-present",
     "check": "npm run lint; npm run build; npm run test",
     "reset": "npm run clean; npm run check",
-    "new-version": "lerna version --no-private --no-git-tag-version --exact && ./scripts/update_relayer_sdk_version.sh",
+    "new-version": "lerna version --no-private --no-git-tag-version --exact && ./scripts/update_relayer_sdk_version.sh && ./scripts/update_web3wallet_version.sh; npm i",
```
Comment (bkrem) [inline package.json]: This will run `npm i` regardless of failure of the previous scripts, is that wanted?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: "20.x"
+          registry-url: "https://registry.npmjs.org"
+      - name: Install dependencies and build 🔧
+        run: npm ci && npm run build
+      - name: Publish package on NPM 📦
+        run: npm run npm-publish:latest
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (bkrem) [inline .github/workflows/publish.yml:21]: You should be able to generate this with granular access here: https://www.npmjs.com/settings/gancho_walletconnect/tokens

Lmk if you can't and I'll take a look 👍

**Review by bkrem — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (ganchoradkov) [reply .github/workflows/publish.yml:21]: created the token & added it to the repo secrets

**Pull request merged**